### PR TITLE
[lsp] Add support for ignored files in ls formatting

### PR DESCRIPTION
**Pull request by charlieegan3**
Adding functionality to manage the contents of ignored files when formatting documents.

Addresses some issues with ignored files missed in https://github.com/StyraInc/regal/pull/837

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -449,6 +449,8 @@ rules:
 
 ### Ignoring Files Globally
 
+**Note**: Ignoring files globally will also disable language server features for those files too.
```
Comment (anderseknert) [inline README.md:452]: Nit for sure, but I don't think you need the ending "too" after having used "also" before :P

**Review by anderseknert — APPROVED** [review]
LGTM! We can extend this to cover more "local" operations later if ever requested.

File: README.md (hunk 1 above)
Comment (charlieegan3) [reply README.md:452]: I made it a little more explicit: "**Note**: Ignoring files will disable most language server features for those files. Only formatting will remain available. Ignored files won't be used for completions, linting, or definitions in other files."

Comment (anderseknert) [reply README.md]: Excellent!

**Pull request merged**